MusicNotationKit/MusicNotationKit/Tuplet.swift

```diff
@@ -117,72 +132,338 @@ public struct Tuplet: NoteCollection {
         return try note(from: fullIndexes)
     }
 
-    public mutating func replaceNote(at index: Int, with note: Note) throws {
-        try replaceNote(at: index, with: note as NoteCollection)
+    public mutating func replaceNote<T: NoteCollection>(at index: Int, with noteCollection: T) throws {
```

Comment:
It seems that `replaceNote` in the general case can be used to replace a `Note` or a `Tuplet` or a range of noteCollection instances. I am not sure if  renaming to `replace(at index, with noteCollection: )` would be descriptive enough?